src/arrayMethodJoin.js

```diff
@@ -4,8 +4,19 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      result += this[i] === null || this[i] === undefined
+        ? '' : String(this[i]);
```

Comment:
In my opinion this would be even more readable:

```js
      if (this[i] !== undefined && this[i] !== null) {
        result += String(this[i]);
      }
```